include/GafferScene/Private/IECoreScenePreview/Renderer.h

```diff
@@ -89,8 +90,12 @@ class IECORESCENE_API Renderer : public IECore::RefCounted
 		IE_CORE_DECLAREMEMBERPTR( Renderer )
 
 		static const std::vector<IECore::InternedString> &types();
-		/// Filename is only used if the renderType is SceneDescription.
-		static Ptr create( const IECore::InternedString &type, RenderType renderType = Batch, const std::string &fileName = "" );
+		/// fileName is only used if the renderType is SceneDescription.
+		/// messageHandler may be provided by the owner of the renderer. If so, all in-render messages should be
+		/// passed to this handler. Message contexts can be left blank if no applicable information is available.
```

Comment:
Are we sure about blank contexts? Would it be better to just have the context be "Arnold" or "3Delight" to help distinguish between those and our own messages?

Reply:
I was mulling that, but would it be somewhat redundant in the render log context to have "Arnold" on every line?

Reply:
Maybe it's OCD, but the idea of an empty context upsets me a bit. And I do think there may be some utility in making it clear when a message comes from the renderer itself vs our own code (geometry converters etc). Not a big deal if you'd like to try without first and see how it goes...

Reply:
I've set the context to the renderer name for 3Dlight/Arnold/Appleseed.